versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoDbStore.java

```
@@ -0,0 +1,141 @@
+ * The MongoDbStore connects to an external MongoDB server.
+ */
+public class MongoDbStore implements Store {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MongoDbStore.class);
+
+  private final String connectionString;
+  private final String databaseName;
+
+  // The client that connects to the MongoDB server.
+  protected MongoClient mongoClient;
+  // The database hosted by the MongoDB server.
+  private MongoDatabase mongoDatabase;
+
+  /**
+   * Creates a store ready for connection to a MongoDB instance.
+   * @param connectionString the URL of the database with which to connect. Eg mongodb://localhost:27017.
+   * @param databaseName the name of the database to retrieve
+   */
+  public MongoDbStore(String connectionString, String databaseName) {
+    this.connectionString = connectionString;
+    this.databaseName = databaseName;
+  }
+
+  /**
+   * Creates a connection to an existing database or creates the database if it does not exist.
+   * Since MongoDB creates databases and collections if they do not exist, there is no need to validate the presence of
+   * either before they are used.
+   */
+  @Override
+  public void start() {
+    mongoClient = MongoClients.create(connectionString);
+    mongoDatabase = mongoClient.getDatabase(databaseName);
+  }
+
+  /**
+   * Closes the connection this manager creates to a database. If the connection is already closed this method has
+   * no effect.
+   */
+  @Override
+  public void close() {
+    if (mongoClient != null) {
+      mongoClient.close();
+      mongoClient = null;
+    }
+  }
+
+  @Override
+  public void load(LoadStep loadstep) throws ReferenceNotFoundException {
+  }
+
+  private List<ListMultimap<String, LoadOp<?>>> paginateLoads(LoadStep loadStep, int size) {
+    List<ListMultimap<String, LoadOp<?>>> paginated = new ArrayList<>();
+    return paginated;
+  }
+
+  @Override
+  public <V> boolean putIfAbsent(ValueType type, V value) {
+    return false;
```

Are these stubbed implementations or unsupported operations? Better to throw an error if they are unsupported.

---

If these are stubbed, then put in a comment to say that is the case.

---

I have added throwing UnsupportedOperationException() exception to all unimplemented methods.